pkg/utils/nodes.go

```diff
@@ -33,3 +34,29 @@ func IsNodeNameValid(r client.Reader, nodeName string) (bool, error) {
 	}
 	return true, nil
 }
+
+// used for making new node object for test and have a unique resourceVersion
+// GetNode returns a node object with the name nodeName based on the nodeType input
+func GetNode(nodeType, nodeName string) *corev1.Node {
+	if nodeType == "control-plane" {
+		return &corev1.Node{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: nodeName,
+			},
+			Spec: corev1.NodeSpec{
+				Taints: []corev1.Taint{
+					{
+						Key:    medik8sLabels.ControlPlaneRole,
+						Effect: corev1.TaintEffectNoExecute,
+					},
+				},
+			},
+		}
+	} else {
+		return &corev1.Node{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: nodeName,
+			},
+		}
+	}
+}
```

Comment:
I think it makes more sense to move this function which is used for unit-testing to another file. Maybe under [`test`](https://github.com/medik8s/fence-agents-remediation/tree/main/test)?

Reply:
Or even having two local function for getting the node object? One in `fenceagentsremediation_controller_test.go` , and one in `utils_test.go`

Reply:
Good point. I would put it into pkg/utils/nodes_test.go

Reply:
But then `controllers/fenceagentsremediation_controller_test.go` could not find the `GetNode` function from a test code `pkg/utils/nodes_test.go`.